benches/store.rs

```diff
@@ -0,0 +1,353 @@
+            b.iter_batched(
+                || {
+                    leaves
+                        .iter()
+                        .enumerate()
+                        .map(|(c, v)| (c.try_into().unwrap(), v.into()))
+                        .collect::<Vec<(u64, Word)>>()
+                },
+                |l| black_box(SimpleSmt::new(63).unwrap().with_leaves(l)),
+                BatchSize::SmallInput,
+            )
+        });
+
+        group.bench_function(
+            BenchmarkId::new("MerkleStore::with_sparse_merkle_tree", size),
+            |b| {
+                b.iter_batched(
+                    || {
+                        leaves
+                            .iter()
+                            .enumerate()
+                            .map(|(c, v)| (c.try_into().unwrap(), v.into()))
+                            .collect::<Vec<(u64, Word)>>()
+                    },
+                    |l| black_box(MerkleStore::new().with_sparse_merkle_tree(l)),
+                    BatchSize::SmallInput,
+                )
+            },
+        );
+    }
+}
+
+/// Benchmarks getting a node on MerkleTree and MerkleStore backends.
+///
+/// Since MerkleTree can only be created when a power-of-two number of elements is used, the sample
+/// sizes are limited to that
+fn get_leaf_merkletree(c: &mut Criterion) {
+    let mut group = c.benchmark_group("get_leaf_merkletree");
+
+    let random_data_size = BATCH_SIZES.into_iter().max().unwrap();
+    let random_data: Vec<RpoDigest> = (0..random_data_size)
+        .map(|_| rand_array::<Felt, 4>().into())
+        .collect();
+
+    for size in BATCH_SIZES {
+        let leaves = &random_data[..size];
+
+        let mtree_leaves: Vec<Word> = leaves.iter().map(|v| v.into()).collect();
+        let mtree = MerkleTree::new(mtree_leaves.clone()).unwrap();
+        let store = MerkleStore::new().with_merkle_tree(mtree_leaves).unwrap();
+        let depth = mtree.depth();
+        let root = mtree.root();
+
+        group.bench_function(BenchmarkId::new("MerkleTree", size), |b| {
+            b.iter_batched(
+                || rand_value::<u64>() % u64::try_from(size).unwrap(),
+                |value| black_box(mtree.get_node(NodeIndex::new(depth, value))),
+                BatchSize::SmallInput,
+            )
+        });
+
+        group.bench_function(BenchmarkId::new("MerkleStore", size), |b| {
+            b.iter_batched(
+                || rand_value::<u64>() % u64::try_from(size).unwrap(),
+                |value| black_box(store.get_node(root, NodeIndex::new(depth, value))),
+                BatchSize::SmallInput,
+            )
+        });
+    }
+}
+
+/// Benchmarks getting an invalid index on the MerkleTree and MerkleStore backends.
+///
+/// Since MerkleTree can only be created when a power-of-two number of elements is used, the sample
+/// sizes are limited to that
+fn get_leaf_merkletree_invalid(c: &mut Criterion) {
```

Comment:
I'm not sure we need this benchmark. I mean it doesn't hurt to have it, but also, this is probably a rather edge case.

Reply:
this was requested here https://github.com/0xPolygonMiden/crypto/pull/97#discussion_r1141613289 , should I remove it?

Reply:
removed. I had understood that the benchmark was for invalid indexes, not for empty hashes, benchmark is updated